### PR TITLE
Region and session lock

### PR DESCRIPTION
**Pull request by liach**
Signed-off-by: liach <liach@users.noreply.github.com>

### CODE REVIEW

File: mappings/net/minecraft/world/storage/RegionFile.mapping [hunk 1]
```diff
@@ -11,9 +11,12 @@ CLASS net/minecraft/class_2861 net/minecraft/world/storage/RegionFile
 	METHOD <init> (Ljava/io/File;Ljava/io/File;Z)V
 		ARG 1 file
 		ARG 2 directory
+		ARG 3 dsync
```
Comment (Runemoro) [inline mappings/net/minecraft/world/storage/RegionFile.mapping]: What does the "d" mean here?

Comment (Runemoro) [reply mappings/net/minecraft/world/storage/RegionFile.mapping]: Just realized this name is from Java.

Comment (Runemoro) [reply mappings/net/minecraft/world/storage/RegionFile.mapping:14]: Just realized this name is from Java.

**Review by Runemoro — APPROVED** [review]

**Comment by modmuss50** [comment]
conflicts please.

File: mappings/net/minecraft/world/gen/StructureAccessor.mapping [hunk 2]
```diff
@@ -0,0 +1,19 @@
+CLASS net/minecraft/class_5138 net/minecraft/world/gen/StructureAccessor
+	METHOD method_26973 addStructureReference (Lnet/minecraft/class_4076;Lnet/minecraft/class_3195;JLnet/minecraft/class_2810;)V
+		ARG 1 pos
+		ARG 2 feature
+		ARG 3 reference
+		ARG 5 holder
+	METHOD method_26974 getStructruesWithChildren (Lnet/minecraft/class_4076;Lnet/minecraft/class_3195;Lnet/minecraft/class_1936;)Ljava/util/stream/Stream;
```
Comment (YanisBft) [inline mappings/net/minecraft/world/gen/StructureAccessor.mapping]:
```suggestion
	METHOD method_26974 getStructuresWithChildren (Lnet/minecraft/class_4076;Lnet/minecraft/class_3195;Lnet/minecraft/class_1936;)Ljava/util/stream/Stream;
```

File: mappings/net/minecraft/world/storage/SerializingRegionBasedStorage.mapping [hunk 3]
```diff
@@ -24,5 +30,11 @@ CLASS net/minecraft/class_4180 net/minecraft/world/storage/SerializingRegionBase
 		ARG 1 pos
 	METHOD method_19295 getOrCreate (J)Ljava/lang/Object;
 		ARG 1 pos
+	METHOD method_20368 update (Lnet/minecraft/class_1923;Lcom/mojang/datafixers/types/DynamicOps;Ljava/lang/Object;)V
+		ARG 1 pos
+		ARG 2 ops
+		ARG 3 data
 	METHOD method_20369 getDataVersion (Lcom/mojang/datafixers/Dynamic;)I
 	METHOD method_20370 save (Lnet/minecraft/class_1923;)V
+	METHOD method_20621 loadNbt (Lnet/minecraft/class_1923;)Lnet/minecraft/class_2487;
```
Comment (YanisBft) [inline mappings/net/minecraft/world/storage/SerializingRegionBasedStorage.mapping:39]: `loadNbt`? Did you turn into Prospector?

**Review by YanisBft — APPROVED** [review]

File: mappings/net/minecraft/world/storage/SerializingRegionBasedStorage.mapping (hunk 3 above)
Comment (liach) [reply mappings/net/minecraft/world/storage/SerializingRegionBasedStorage.mapping:39]: I recall there is a specific reason I named this, let me look into it real quick (like this is not copying wholesale but some sort of selective reading)

Comment (liach) [reply mappings/net/minecraft/world/storage/SerializingRegionBasedStorage.mapping:39]: Oh, this is not a `fromNbt` but rather getting nbt from disk (loaded by the storage io worker that is a field of it, calls its `getNbt` and wraps the call in an exception check). What name do you suggest?

Comment (YanisBft) [reply mappings/net/minecraft/world/storage/SerializingRegionBasedStorage.mapping:39]: I'm fine with this name, I was just surprised as the current convention is to use "tag(s)" over "nbt"

**Pull request merged**